lib/persist-state.h

```diff
@@ -97,6 +97,7 @@ gboolean persist_state_remove_entry(PersistState *self, const gchar *persist_nam
 
 gchar *persist_state_lookup_string(PersistState *self, const gchar *key, gsize *length, guint8 *version);
 gboolean persist_state_rename_entry(PersistState *self, const gchar *old_key, const gchar *new_key);
+void persist_state_copy_entry_value(PersistState *self, const PersistEntryHandle from, PersistEntryHandle to, gsize size);
```

Comment:
I think we should use `const gchar *old_key, const gchar *new_key` instead, so you could move everything into the persist level and make sure that `size` will be the same for `from` and `to`.